rsciio/jobin_yvon/api.py

```diff
@@ -0,0 +1,819 @@
+            del self.original_metadata["experimental_setup"]["ND Filter"]
+        except KeyError:
+            pass  # pragma: no cover
+
+    def get_original_metadata(self):
+        """Extracts metadata from file."""
+        self.original_metadata = {}
+        for child in self._metadata_root:
+            id = self._get_id(child)
+            if id == "0x7CECDBD7":
+                date = child
+            if id == "0x8716361":
+                metadata = child
+            if id == "0x7C73E2D2":
+                file_specs = child
+
+        ## setup tree structure original_metadata -> date{...}, experimental_setup{...}, file_information{...}
+        ## based on structure in file
+        self._get_metadata_values(date, "date")
+        self._get_metadata_values(metadata, "experimental_setup")
+        self._get_metadata_values(file_specs, "file_information")
+        try:
+            self.original_metadata["experimental_setup"][
+                "measurement_type"
+            ] = self._measurement_type
+        except AttributeError:
+            pass  # pragma: no cover
+        try:
+            self.original_metadata["experimental_setup"]["title"] = self._title
+        except AttributeError:
+            pass  # pragma: no cover
+        try:
+            self.original_metadata["experimental_setup"][
+                "rotation angle (rad)"
+            ] = self._angle
+        except AttributeError:
+            pass
+        self._clean_up_metadata()
+
+    def _set_signal_type(self, xml_element):
+        """Sets signal type and units based on metadata from file.
+
+        Extra method, because this information is stored seperate from the rest of the metadata.
+
+        Parameters
+        ----------
+        xml_element: xml.etree.ElementTree.Element
+            Head level metadata element.
+        """
+        for child in xml_element:
+            id = self._get_id(child)
+            ## contains also intensity-minima/maxima-values for each data-row (ignored by this reader)
+            if id == "0x6D707974":
+                self.original_metadata["experimental_setup"]["signal type"] = child.text
+            if id == "0x7C696E75":
+                self.original_metadata["experimental_setup"][
+                    "signal units"
+                ] = child.text
+
+    def _get_scale(self, array, name):
+        """Get scale for navigation/signal axes.
+
+        Furthermore, checks whether the axis is uniform. Throws warning when this not the case.
+        This check is performed by comparing the difference between the first 2 and last 2 values.
+        The decision to use a non-uniform/uniform data-axis is left to the user
+        (use_uniform_wavelength_axis).
+
+        Parameters
+        ----------
+        array: np.ndarray
+            Contains the axes data points.
+        name: Str
+            Name of the axis.
+        """
+        scale = np.abs(array[0] - array[-1]) / (array.size - 1)
```

Comment:
I like the approach of loading as uniform axis and warn the user of possible non-uniformity. 

However, I would use a regression method such as numpy.polyfit in order to minimize the linearized axis error:
`[scale, offset] = np.polyfit(np.arange(array.size),array,1)`

Reply:
An advantage of the method I used is that the boundaries match exactly.
Moreover, the offset obtained from the fit is wrong when the axis contains negative values.
For example there is a test file with navigation axis (-2, 0, 2).

I personally wouldn't use the fit for the offset calculation, but I could imagine using a fit for the scale.

Reply:
Indeed, the only additional degree of freedom of the fit should be that the first and last point are not fixed. So if we want to keep their positions, there is no advantage of the fit.

Reply:
> An advantage of the method I used is that the boundaries match exactly.

> Indeed, the only additional degree of freedom of the fit should be that the first and last point are not fixed. So if we want to keep their positions, there is no advantage of the fit.

I don't see any virtue in ensuring that first/last points match exactly but if that's a concern to you I guess you have good reasons.

> Moreover, the offset obtained from the fit is wrong when the axis contains negative values.
For example there is a test file with navigation axis (-2, 0, 2).

Sounds odd to me. Running this gives
```python
np.polyfit(x=np.arange(3),y=[-2,0,2],deg=1)
#out: array([ 2., -2.])
```
Which is exactly a scale of 2 and an offset of -2

> I personally wouldn't use the fit for the offset calculation, but I could imagine using a fit for the scale.

Definitely this would induce more error than what is currently implemented (at lease in some cases)

You do as you please :)

Reply:
> Which is exactly a scale of 2 and an offset of -2

I made a mistake here. Yes it should also work in this case.

I use a fit now to get the scale/offset of the signal axis.